Modernize Docs

jordanjones243 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Checklist for PR Reviewer:

#### Documentation Accuracy
- [x] Do all explanations in the API table convey the feature they are referencing in an understandable way?
    - [x] Are all `String` properties displaying all of their "options"?
- [x] Are all necessary API features documented? Are there any missing that you believe should be added?
- [x] Does the component description and use case section convey what the component is all about? Is there too much or too little information?
- [x] Ensure only NECESSARY examples are on the `index` page (i.e. only the most "basic" example(s) of component)
- [x] Are the API examples more or less appearance in the same order as they are in on the API table?
- [x] Ensure custom registration example is now on `install` page and renders as expected
- [x] Do all functional JS examples still work?

#### Code Changes
- [x] Ensure all example files are rename from camel case to kebab case (ex. `myExample.html` -> `my-example.html`)
- [x] Ensure no backticks appear in example header names
- [x] Is there a `perf` commit to trigger a release, as expected?

#### Visual Checks
- [x] Ensure all `See Code` accordions actually render the code blocks under their respective examples
- [x] Do all examples with multiple components in them have sufficient spacing?

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

DukeFerdinand reviewed (changes requested):
Seems like the `bottom` option for the `placement` attribute is not working as expected?

<img width="1129" height="462" alt="Screenshot 2026-01-05 at 10 30 31 AM" src="https://github.com/user-attachments/assets/00f343bf-1da7-4707-b695-47596e7afe8b" />

jordanjones243 committed:
perf: modernize component example documentation #118
example code updated to reflect latest Auro element APIs
example documentation layout improved
custom registration updated and moved to install page
core element descriptions updated
properties now listed alphabetically
doc build system now uses the latest version of auro-cli
package json configuration updated to support latest auro-cli
additional packages dependencies updated to latest versions
